scipy/fft/_pocketfft/basic.py

```diff
@@ -0,0 +1,262 @@
+_default_workers = 1
+
+def _datacopied(arr, original):
+    """
+    Strict check for `arr` not sharing any data with `original`,
+    under the assumption that arr = asarray(original)
+    """
+    if arr is original:
+        return False
+    if not isinstance(original, np.ndarray) and hasattr(original, '__array__'):
+        return False
+    return arr.base is None
+
+
+def _fix_shape(x, shape, axes):
+    """Internal auxiliary function for _raw_fft, _raw_fftnd."""
+    must_copy = False
+
+    # Build an nd slice with the dimensions to be read from x
+    index = [slice(None)]*x.ndim
+    for n, ax in zip(shape, axes):
+        if x.shape[ax] >= n:
+            index[ax] = slice(0, n)
+        else:
+            index[ax] = slice(0, x.shape[ax])
+            must_copy = True
+
+    index = tuple(index)
+
+    if not must_copy:
+        return x[index], False
+
+    s = list(x.shape)
+    for n, axis in zip(shape, axes):
+        s[axis] = n
+
+    z = np.zeros(s, x.dtype)
+    z[index] = x[index]
+    return z, True
+
+
+def _fix_shape_1d(x, n, axis):
+    if n < 1:
+        raise ValueError(
+            "invalid number of data points ({0}) specified".format(n))
+
+    return _fix_shape(x, (n,), (axis,))
+
+
+def _normalization(norm, forward):
+    """Returns the pypocketfft normalization mode from the norm argument"""
+
+    if norm is None:
+        return pfft.norm_t.none if forward else pfft.norm_t.size
+
+    if norm == 'ortho':
+        return pfft.norm_t.ortho
+
+    raise ValueError(
+        "Invalid norm value {}, should be None or \"ortho\".".format(norm))
+
+
+def fft(x, n=None, axis=-1, norm=None, overwrite_x=False):
+    """ Return discrete Fourier transform of real or complex sequence. """
+    tmp = np.asarray(x)
+    overwrite_x = overwrite_x or _datacopied(tmp, x)
+    norm = _normalization(norm, True)
+
+    if n is not None:
+        tmp, copied = _fix_shape_1d(tmp, n, axis)
+        overwrite_x = overwrite_x or copied
+    elif tmp.shape[axis] < 1:
+        raise ValueError("invalid number of data points ({0}) specified"
+                         .format(tmp.shape[axis]))
+
+    return pfft.fftn(tmp, (axis,), norm, overwrite_x, _default_workers)
+
+
+def ifft(x, n=None, axis=-1, norm=None, overwrite_x=False):
+    """
+    Return discrete inverse Fourier transform of real or complex sequence.
+    """
+    tmp = np.asarray(x)
+    overwrite_x = overwrite_x or _datacopied(tmp, x)
+    norm = _normalization(norm, False)
+
+    if n is not None:
+        tmp, copied = _fix_shape_1d(tmp, n, axis)
+        overwrite_x = overwrite_x or copied
+    elif tmp.shape[axis] < 1:
+        raise ValueError("invalid number of data points ({0}) specified"
+                         .format(tmp.shape[axis]))
+
+    return pfft.ifftn(tmp, (axis,), norm, overwrite_x, _default_workers)
+
+
+def rfft(x, n=None, axis=-1, norm=None, overwrite_x=False):
+    """
+    Discrete Fourier transform of a real sequence.
```

Comment:
I guess these docstrings will be expanded to the same level of details the old ones in a different PR?

Reply:
The user-facing API is in `scipy/fft/basic.py` which has the full docstrings.

Reply:
See the [rendered docs](https://scipy.github.io/devdocs/generated/scipy.fft.fft.html#scipy.fft.fft)